fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -30,8 +33,12 @@ void MessageLoopFuchsia::WakeUp(fml::TimePoint time_point) {
     due_time = zx::nsec((time_point - now).ToNanoseconds());
   }
 
-  auto status = async::PostDelayedTask(
-      loop_.dispatcher(), [this]() { RunExpiredTasksNow(); }, due_time);
+  std::scoped_lock lock(task_mutex_);
```

Comment:
Does this timing issue exist on other platforms?  MessageLoopLinux::WakeUp does not do any synchronization, for example

Reply:
Actually, all of the entrypoints in MessageLoopTaskQueues are already guarded by locks.  This is why none of the other platform implementations do any locking in WakeUp.

So, we added this lock for no reason